fix: update chat panel title on resetting chat

abeatrix opened the pull request:
CLOSE https://github.com/sourcegraph/cody/issues/2785

- Replaced command name `cody.chat.restart` with `cody.chat.panel.reset`
- Extracted reset logic from ChatPanelsManager into a `cody.chat.panel.reset` command 
- Moved the old command registration step from main to ChatManager
- Added `cody.chat.panel.refresh` command to update provider map
  - [x] `clearAndRestartSession` now reset the chat panel title (fixed in this PR)
  - [x] clicking on the old chat session in the history sidebar should open it in a new chat panel (fixed in https://github.com/sourcegraph/cody/pull/2768)
- Added e2e test to cover this change

### Before

#### Reproduce Steps

1. Ask Cody a question
2. Clear your current chat panel by clicking on the "Restart" button  in your editor title
3. Ask Cody a question again in the same chat panel which should be cleared
4. Click on the previous chat session on your chat history sidebar

https://github.com/sourcegraph/cody/assets/68532117/4455241d-10c0-411f-a0ab-7e6ea19fe7b5

## Test plan

<!-- Required. See https://sourcegraph.com/docs/dev/background-information/testing_principles. -->

1. Ask Cody a question
2. Clear your current chat panel by clicking on the "Restart" button  in your editor title
3. Ask Cody a question again in the same chat panel which should be cleared
4. Click on the previous chat session on your chat history sidebar
5. You should be able to open a new chat panel with previous chat session displayed

### After

https://github.com/sourcegraph/cody/assets/68532117/d36bf666-b76a-4a84-a7d7-b5f806501eeb

sqs commented on vscode/src/chat/chat-view/ChatManager.ts:
```diff
@@ -78,6 +78,8 @@ export class ChatManager implements vscode.Disposable {
             vscode.commands.registerCommand('cody.chat.history.edit', async item => this.editChatHistory(item)),
             vscode.commands.registerCommand('cody.chat.panel.new', async () => this.createNewWebviewPanel()),
             vscode.commands.registerCommand('cody.chat.panel.restore', (id, chat) => this.restorePanel(id, chat)),
+            vscode.commands.registerCommand('cody.chat.panel.reset', async id => this.chatPanelsManager.resetPanel(id)),
```
async keyword is not needed

sqs commented on vscode/src/chat/chat-view/ChatPanelsManager.ts:
```diff
@@ -262,17 +264,33 @@ export class ChatPanelsManager implements vscode.Disposable {
+     */
+    public async resetPanel(id?: string): Promise<void> {
+        logDebug('ChatPanelsManager', 'resetPanel')
+        // NOTE: the editor title button click returns a uri instead of id string
```
Recommend to filter that out in the registerCommand call above so that this function can just be concerned with the inputs that are within its scope.

So it would be like

```
vscode.commands.registerCommand('cody.chat.panel.reset', idOrUri => this.chatPanelsManager.resetPanel(typeof idOrUri === 'string' ? idOrUri : undefined))
```

And then explain why there - instead of saying "returns a uri" it is "When this command is invoked by clicking on the button on the editor title [as opposed to what?], the arg is a vscode.Uri. When invoked by ___, it is a string.".

sqs commented on vscode/src/chat/chat-view/ChatPanelsManager.ts:
```diff
@@ -291,6 +309,16 @@ export class ChatPanelsManager implements vscode.Disposable {
         }
     }
 
+    // Refresh the provider map
```
For a docstring, use jsdoc syntax `/** ...`.

Why would someone need to refresh the provider map?

sqs commented on vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:
```diff
@@ -538,11 +538,19 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
-    public handleChatTitle(title: string): void {
-        this.chatModel.setChatTitle(title)
+    public handleChatTitle(title = 'New Chat'): void {
+        // setChatTitle handles title changed by user
```
This comment should be in the jsdoc docstring for the method imo

sqs commented on vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:
```diff
@@ -538,11 +538,19 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
 
-    public handleChatTitle(title: string): void {
-        this.chatModel.setChatTitle(title)
+    public handleChatTitle(title = 'New Chat'): void {
```
factor out `"New Chat"` to a const

sqs reviewed (approved):
Minor changes

arafatkatze commented on vscode/test/e2e/chat-history.test.ts:
```diff
@@ -0,0 +1,54 @@
+import { test } from './helpers'
+
+test('shows chat history in sidebar and update chat panel correctly', async ({ page, sidebar }) => {
+    // Sign into Cody
```
Sick!!!

arafatkatze commented on vscode/package.json:
```diff
@@ -465,7 +465,7 @@
-        "command": "cody.chat.restart",
+        "command": "cody.chat.panel.reset",
         "category": "Cody",
         "title": "Restart Chat Session...",
```
<img width="305" alt="image" src="https://github.com/sourcegraph/cody/assets/11155207/1f2c2223-117c-4460-93a5-7caa73913a46">
You just reset and NOT restart.
Also, you don't erase the chat history too so change the wording around it.

arafatkatze commented on vscode/src/chat/chat-view/ChatPanelsManager.ts:
```diff
@@ -262,17 +264,33 @@ export class ChatPanelsManager implements vscode.Disposable {
+            // Show confirmation dialog
+            const confirmation = await vscode.window.showWarningMessage(
+                'Restart Chat Session',
+                { modal: true, detail: 'Restarting the chat session will erase the chat transcript.' },
```
Something about this wording shall change coz you dont erase

abeatrix replied:
Yea I agree that this seems to be unnecessary and looks misleading, will remove. Thank you!

arafatkatze reviewed (approved):
Approving if [this](https://github.com/sourcegraph/cody/pull/2786/files#r1456926527) is solved

abeatrix commented:
The initial issue was fixed by beyang in https://github.com/sourcegraph/cody/pull/2768 so I just updated this PR with a minor fix with some cleanups, and added a e2e test to cover the fixes.

Thank you so much for the reviews!
